USWDS-Site: Settings documentation

amyleadem opened the pull request:
## Summary
Removed redundant instructions for setting up the Sass file structure from the settings page.

## Related issue
Closes https://github.com/uswds/uswds-site/issues/1701

Recommend releasing this update with the Phase 2 compile update in PR https://github.com/uswds/uswds-site/pull/1766

## Preview link
Preview link: [Settings page](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.app.cloud.gov/preview/uswds/uswds-site/al-settings/documentation/settings/)

## Problem statement
Providing the same instructions in two different locations can cause confusion for the user and requires double maintenance. Additionally, users looking to configure their settings will receive more information than needed.

## Solution
Replacing the instructions for setting up Sass files with a link to the Getting Started guide allows the user to focus on only configuring the settings. 

## Testing and review
To review:
- Proofread for errors
- Confirm that instructions makes sense 
- Confirm that instructions are comprehensive

Before opening this PR, make sure you’ve done whichever of these applies to you:

- [x] Confirm that this code follows the [18F Front End Coding Style Guide](https://pages.18f.gov/frontend/) and [Accessibility Guide](https://pages.18f.gov/accessibility/checklist/).
- [x] Run `git pull origin [base branch]` to pull in the most recent updates from your base and check for merge conflicts. (Often, the base branch is `develop`).
- [x] Run `npm test` and confirm that all tests pass.
- [x] Run your code through [HTML_CodeSniffer](http://squizlabs.github.io/HTML_CodeSniffer/) and make sure it’s error free.

amyleadem commented:
@bonnieAcameron @mejiaj 
For this issue, I discovered that the instructions on the settings page had a lot of overlap with [Phase 2, Step 1 of the getting started guide](https://designsystem.digital.gov/documentation/getting-started/developers/phase-two-compile/). I took a stab at restructuring the content for clarity so that the focus is entirely on settings configuration here, then point users to file setup instructions if they need it. 

I know this is larger than the scope of the original issue, so let me know if the solution should be reined in. Let me know if you have any thoughts/opinions!

bonnieAcameron reviewed (changes requested):
Hi Amy! I edited the file with a few of my suggestions & a grammar update, but this is an excellent update to the settings page.

mejiaj commented on pages/documentation/settings.md:
```diff
@@ -5,6 +5,8 @@ title: Settings
 category: How to use USWDS
 lead: USWDS settings variables tell the design system how to build.
 subnav:
+- text: Configuring USWDS settings
+  href: '#configuring-uswds-settings'
```
Is this missing "Custom"?

```suggestion
- text: Configuring Custom USWDS settings
  href: '#configuring-custom-uswds-settings'
```

mejiaj commented on pages/documentation/settings.md:
```diff
@@ -19,17 +21,37 @@ subnav:
+Note that the settings variables in this module inform both general and component customizations.
+
+### What to include in your configuration
+Include only USWDS settings that you wish to modify in this configuration. These items should be comma-separated and given values of the appropriate type.
```
**Optional**, but what is the appropriate type? Maybe specify that it's usually design tokens again.

amyleadem replied:
@bonnieAcameron @mejiaj My word brain is moving slow today. Does this add clarity? 

```suggestion
Include only USWDS settings that you wish to modify in this configuration. These items should be comma-separated and given values of the appropriate type, as identified in the tables below.
```

mejiaj commented on pages/documentation/settings.md:
```diff
@@ -19,17 +21,37 @@ subnav:
+### What to include in your configuration
+Include only USWDS settings that you wish to modify in this configuration. These items should be comma-separated and given values of the appropriate type.
+
+Please note that this configuration accepts only current USWDS settings variables &mdash; adding anything else here will result in an error.
```
Maybe add site-note here to make it standout?

amyleadem replied:
There is a more specific site-note about this at the [bottom of this file](https://github.com/uswds/uswds-site/pull/1755/files#diff-d26ac88e392b8b5ae6f8fd8fc17fa7ace2877ed30ea067822dab4e3190a19b32R86-R87) - should it be moved up? Is it redundant? I lean towards putting specific error notes at the bottom of the copy and more casual/generic guidance up top, but I am very open to input here.

amyleadem replied:
Update: I kept the current format of having the error site-note at the bottom of the text because I wanted this copy to remain very straightforward. Let me know if you'd like to see something different.

mejiaj replied:
No, it makes sense. Thanks for taking another look.

mejiaj reviewed (changes requested):
Nice job! This introduction seems clear and straightforward. Added some comments and questions on a few small things.

amyleadem commented:
> Line 25-If we're specifically talking about the USWDS Design System, then we capitalize it: the Design System
> Line 25, 44-instead of saying "tables below," (directional language isn't best practice for accessibility) can we use active voice to say, you can find USWDS settings and their default values in the [USWDS settings tables](#general-settings)

@bonnieAcameron Great feedback! I've updated these items in [a5c672f](https://github.com/uswds/uswds-site/pull/1755/commits/a5c672ff17fd57b30fc8da5b49433ae86d1895ec)

mejiaj reviewed (approved):
Good changes, I like the added clarity. Thanks @amyleadem!

bonnieAcameron commented on pages/documentation/settings.md:
````diff
@@ -59,11 +83,6 @@ If you include settings right in your Sass entry point, it would look something
 @forward 'uswds-theme-custom-styles';
 ```
 
-In general, when including settings and custom code, you just need to follow this order of operations:
````
Wholeheartedly agree with the 62-66 lines deletion

bonnieAcameron commented on pages/documentation/settings.md:
```diff
@@ -19,17 +21,37 @@ subnav:
+### Where to include your configuration
+Full instructions for setting up Sass files for USWDS can be found in [Phase 2 of the Getting Started for Developers]({{ site.baseurl }}/documentation/getting-started/developers/phase-two-compile/#step-1-set-up-your-projects-sass-entry-point) guide.
+
+The important thing to note is that your settings configuration must be loaded **_before_ including the main USWDS Sass**. We've created some example setups to give you an idea of how this works:
```
Line 51: I'm a bit confused about the order of operations, here. Where are the steps to include the main USWDS sass? I feel a bit nervous that if I get to this step I might have already included the main uswds sass without knowing/accidentally. Could we use active voice and make the order very clear.  Say something like, 

To make sure everything imports properly:

1. load your settings configuration (hyperlinked)
2. include the main Uswds sass (hyperlinked if appropriate)

We've created some example setups... [...]

[optional: explain what error they could see if they get this order wrong in case people are searching the site by error]

bonnieAcameron reviewed (changes requested):
Reviewed for clarity, plain language (within reason) & typos.

Had a few ideas/clarifications, but other than that, fantastic changes.

amyleadem commented:
@bonnieAcameron 
Thanks for the notes! They helped me back up and make fewer assumptions.

I added some items to clarify in these commits: [2073839](https://github.com/uswds/uswds-site/pull/1755/commits/20738393692c9f5e1e2fd03c409a762962068f8e) and  [840bffd](https://github.com/uswds/uswds-site/pull/1755/commits/840bffd589096200e8a3547d573ba4137d8f3055). 

- Restructured the `Where to include your configuration section` for clarity
- Replaced `Example 1` and `Example 2` with more descriptive names
- Added more information up top about where this configuration should go
- Added links to relevant Sass pages so users can get a deeper understanding if desired (I am guessing at least some users are still getting the hang it)

amyleadem commented on pages/documentation/settings.md:
```diff
@@ -19,20 +21,47 @@ subnav:
+## Configuring custom USWDS settings
+With the introduction of [Sass modules](https://sass-lang.com/blog/the-module-system-is-launched) in USWDS 3, you can now create a custom configuration of USWDS by loading a single [`@use...with()`](https://sass-lang.com/documentation/at-rules/use#configuration) rule into your [Sass entry point]({{ site.baseurl }}/documentation/getting-started/developers/phase-two-compile/#step-1-set-up-your-proj ...
+
+All USWDS settings are defined in the `uswds-core` module, which means the USWDS settings configuration rule is `@use "uswds-core" with ()`.
```
Added these paragraphs because I didn't want to make the assumption that users know how `@use` works and where this configuration should live. It feels a bit clunky to me, so if you have any polish to rub on it, please let me know!

bonnieAcameron replied:
I think this looks awesome!! The only part where I feel it's a bit clunky (I probably wouldn't have noticed if you hadn't said anything) is line 30, which could potentially move to the start of line 28? It's like a bit of context you may need before following the instructions.

This is so beautiful and clear I almost feel like I could do it!

amyleadem replied:
Thanks @bonnieAcameron! 

As far as moving line 30, I think it needs to stay in the current order. Here's why:
One other way to say `@use...with()` is `@use <url> with()`. So line 28 is explaining generally what the `@use <url> with()` rule is and provides links to more information about how that rule works (This was done because this technique is relatively new and can be confusing even to users who are familiar with Sass). Then line 30 gives the specific USWDS `<url>` ("uswds-core") that is needed to complete the rule. 

If you see another way of writing this, please let me know!

bonnieAcameron reviewed (approved):
Looks great, Amy!

thisisdano reviewed (approved):
Nice improvement here
